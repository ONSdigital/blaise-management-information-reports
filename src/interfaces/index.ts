interface InterviewerCallHistoryReportData {
    questionnaire_name: string
    serial_number: string
    call_start_time: string
    dial_secs: string
    call_result: string
}

interface AppointmentResourcePlanningReportData {
    questionnaire_name: string
    appointment_time: string
    appointment_language: string
    total: number
}

export type {InterviewerCallHistoryReportData, AppointmentResourcePlanningReportData};
