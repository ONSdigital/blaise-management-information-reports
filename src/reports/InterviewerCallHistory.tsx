import React, {ReactElement, useState} from "react";
import {Link} from "react-router-dom";
import {ONSButton, ONSPanel} from "blaise-design-system-react-components";
import FormTextInput from "../form/TextInput";
import Form from "../form";
import {requiredValidator} from "../form/FormValidators";
import {getReport} from "../utilities/http";
import {ErrorBoundary} from "../Components/ErrorHandling/ErrorBoundary";
import {ONSDateInput} from "../Components/ONSDesignSystem/ONSDateInput";

function InterviewerCallHistory(): ReactElement {
    const [buttonLoading, setButtonLoading] = useState<boolean>(false);
    const [interviewerID, setInterviewerID] = useState<string>("");
    const [startDate, setStartDate] = useState<Date>(new Date());
    const [endDate, setEndDate] = useState<Date>(new Date());
    const [message, setMessage] = useState<string>("");
    const [listReportData, setListReportData] = useState<any[]>([]);

    async function runReport(formData: any) {
        console.log(formData);
        setInterviewerID(formData.interviewer);
        formData.start_date = startDate;
        formData.end_date = endDate;

        console.log(`runReport ${formData}`);

        const [success, list] = await getReport(formData);

        if (!success) {
            setMessage("Error running report");
            return;
        }

        console.log(list);
        setListReportData(list);

    }

    return (
        <>
            <p className="u-mt-m">
                <Link to={"/"}>Previous</Link>
            </p>
            <h1>Run interviewer call history report</h1>
            <ONSPanel hidden={(message === "")} status="error">
                {message}
            </ONSPanel>
            <Form onSubmit={(data) => runReport(data)}>
                <p>
                    <FormTextInput
                        name="interviewer"
                        validators={[requiredValidator]}
                        label={"Interviewer ID"}
                    />
                </p>
                <ONSDateInput
                    label={"Start Date"}
                    date={startDate}
                    onChange={(date) => setStartDate(date)}
                />
                <br/>
                <ONSDateInput
                    label={"End Date"}
                    date={endDate}
                    onChange={(date) => setEndDate(date)}
                />
                <br/>
                <br/>
                <ONSButton
                    label={"Run"}
                    primary={true}
                    loading={buttonLoading}
                    submit={true}/>
            </Form>
            <br/>

            <ErrorBoundary errorMessageText={"Failed to load"}>
                {
                    listReportData && listReportData.length > 0
                        ?
                        <table id="batches-table" className="table ">
                            <thead className="table__head u-mt-m">
                            <tr className="table__row">
                                {/*
                                <th scope="col" className="table__header ">
                                    <span>Interviewer ID</span>
                                </th>
                                */}
                                <th scope="col" className="table__header ">
                                    <span>Questionnaire</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Serial Number</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Call Start Time</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Call Length</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Interviews</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Call Result</span>
                                </th>
                            </tr>
                            </thead>
                            <tbody className="table__body">
                            {
                                listReportData.map((batch: any) => {
                                    return (
                                        <tr className="table__row" key={batch.name}
                                            data-testid={"batches-table-row"}>
                                            {/*
                                            <td className="table__cell ">
                                                {batch.interviewer}
                                            </td>
                                            */}
                                            <td className="table__cell ">
                                                {batch.questionnaire_name}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.serial_number}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.call_start_time}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.dial_secs}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.number_of_interviews}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.call_result}
                                            </td>
                                        </tr>
                                    );
                                })
                            }
                            </tbody>
                        </table>
                        :
                        <br/>
                }
            </ErrorBoundary>

        </>
    );
}

export default InterviewerCallHistory;
