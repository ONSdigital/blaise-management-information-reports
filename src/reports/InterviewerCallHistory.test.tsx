import {ReportData} from "../interfaces";
import "@testing-library/jest-dom";
import flushPromises, {mock_fetch_requests} from "../tests/utils";
import {createMemoryHistory} from "history";
import {cleanup, render, waitFor} from "@testing-library/react";
import {Router} from "react-router";
import InterviewerCallHistory from "./InterviewerCallHistory";
import {act} from "react-dom/test-utils";
import {fireEvent, screen} from "@testing-library/dom";
import React from "react";

const reportDataReturned: ReportData[] = [

    {
        questionnaire_name: "LMS2101_AA1",
        serial_number: "1337",
        call_start_time: "Sat, 01 May 2021 10:00:00 GMT",
        dial_secs: "61",
        number_of_interviews: "42",
        call_result: "Busy",
    }];

const mock_server_responses_with_data = (url: string) => {
    console.log(url);
    if (url.includes("/api/reports/interviewer-call-history")) {
        return Promise.resolve({
            status: 200,
            json: () => Promise.resolve(reportDataReturned),
        });
    } else if (url.includes("/api/reports/call-history-status")) {
        return Promise.resolve({
            status: 200,
            json: () => Promise.resolve({"last_updated": "Tue, 01 June 2021 10:00:00 GMT"}),
        });
    }
};

const mock_server_responses_without_data = (url: string) => {
    console.log(url);
    if (url.includes("/api/reports/interviewer-call-history")) {
        return Promise.resolve({
            status: 200,
            json: () => Promise.resolve(""),
        });
    } else if (url.includes("/api/reports/call-history-status")) {
        return Promise.resolve({
            status: 200,
            json: () => Promise.resolve(""),
        });
    }
};

describe("interviewer call history report with data", () => {

    beforeEach(() => {
        mock_fetch_requests(mock_server_responses_with_data);
    });

    it("matches snapshot", async () => {
        const history = createMemoryHistory();

        jest.useFakeTimers("modern");
        jest.setSystemTime(new Date("2012-10-10"));

        const wrapper = render(
            <Router history={history}>
                <InterviewerCallHistory/>
            </Router>
        );

        jest.useRealTimers();

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("renders correctly", async () => {
        const history = createMemoryHistory();

        await act(async () => {
            render(
                <Router history={history}>
                    <InterviewerCallHistory/>
                </Router>
            );
        });

        expect(screen.queryByText("Report data last updated: 01/06/2021 11:00:00")).toBeInTheDocument();
        expect(screen.queryByText(/Run interviewer call history report/)).toBeInTheDocument();
        expect(screen.queryByText(/Interviewer ID/)).toBeInTheDocument();
        expect(screen.queryByText(/Start Date/)).toBeInTheDocument();
        expect(screen.queryByText(/End Date/)).toBeInTheDocument();

        fireEvent.input(screen.getByLabelText(/Interviewer ID/i), {
            target: {
                value:
                    "ricer"
            }
        });

        await fireEvent.click(screen.getByTestId(/submit-call-history-form-button/i));

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(/LMS2101_AA1/)).toBeDefined();
            expect(screen.getByText(/1337/)).toBeDefined();
            // TODO: Timezones are terrible
            // expect(screen.getByText("01/05/2021 11:00:00")).toBeDefined();
            expect(screen.getByText(/01:01/)).toBeDefined();
            expect(screen.getByText(/42/)).toBeDefined();
            expect(screen.getByText(/Busy/)).toBeDefined();
        });

    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});

describe("interviewer call history report without data", () => {

    beforeEach(() => {
        mock_fetch_requests(mock_server_responses_without_data);
    });

    it("matches snapshot", async () => {
        const history = createMemoryHistory();

        jest.useFakeTimers("modern");
        jest.setSystemTime(new Date("2012-10-10"));

        const wrapper = render(
            <Router history={history}>
                <InterviewerCallHistory/>
            </Router>
        );

        jest.useRealTimers();

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("renders correctly", async () => {
        const history = createMemoryHistory();

        await act(async () => {
            render(
                <Router history={history}>
                    <InterviewerCallHistory/>
                </Router>
            );
        });

        expect(screen.queryByText(/Run interviewer call history report/)).toBeInTheDocument();
        expect(screen.queryByText(/Interviewer ID/)).toBeInTheDocument();
        expect(screen.queryByText(/Start Date/)).toBeInTheDocument();
        expect(screen.queryByText(/End Date/)).toBeInTheDocument();

        fireEvent.input(screen.getByLabelText(/Interviewer ID/i), {
            target: {
                value:
                    "ricer"
            }
        });

        await fireEvent.click(screen.getByTestId(/submit-call-history-form-button/i));

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.queryByText(/No data found for parameters given./)).toBeInTheDocument();
        });

    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});
