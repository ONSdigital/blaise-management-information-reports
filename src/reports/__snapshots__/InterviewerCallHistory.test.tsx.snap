// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interviewer call history report with data matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
      >
        <ol
          class="breadcrumb__items u-fs-s"
        >
          <li
            class="breadcrumb__item"
            id="breadcrumb-0"
          >
            <a
              class="breadcrumb__link"
              href="/"
            >
              Back
            </a>
            <svg
              class="svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
        </ol>
      </nav>
      <main
        class="page__main u-mt-s"
        id="main-content"
      >
        <h1
          class="u-mb-m"
        >
          Run interviewer call history report
        </h1>
        <p
          aria-live="polite"
          class="u-fs-s u-mt-s"
        >
          Data in this report was last updated: 
          <b>
            <time
              datetime="2000-01-01T10:00:00.000Z"
              title="2000-01-01 10:00"
            >
              2 days ago
            </time>
             (01/01/2000 10:00:00)
          </b>
        </p>
        <p
          class="u-fs-s"
        >
          Data in this report only goes back to the last 12 months.
        </p>
        <form
          action="#"
        >
          <fieldset
            class="fieldset"
          >
            <legend
              class="fieldset__legend"
            >
              Select survey
            </legend>
            <div
              class="radios__items"
              id="Survey TLA"
            >
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    checked=""
                    class="radio__input js-radio"
                    id="all"
                    initial_value="undefined"
                    name="Survey TLA"
                    type="radio"
                    value="undefined"
                  />
                  <label
                    class="radio__label "
                    for="all"
                    id="all-label"
                  >
                    Show all surveys
                  </label>
                </span>
              </p>
              <br />
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    class="radio__input js-radio"
                    id="lms"
                    initial_value="undefined"
                    name="Survey TLA"
                    type="radio"
                    value="lms"
                  />
                  <label
                    class="radio__label label--with-description"
                    for="lms"
                    id="lms-label"
                  >
                    LMS
                    <span
                      class="label__description radio__label--with-description"
                      id="white-label-description-hint"
                    >
                      Labour Market Survey
                    </span>
                  </label>
                </span>
              </p>
              <br />
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    class="radio__input js-radio"
                    id="opn"
                    initial_value="undefined"
                    name="Survey TLA"
                    type="radio"
                    value="opn"
                  />
                  <label
                    class="radio__label label--with-description"
                    for="opn"
                    id="opn-label"
                  >
                    OPN
                    <span
                      class="label__description radio__label--with-description"
                      id="white-label-description-hint"
                    >
                      Opinions and Lifestyle Survey
                    </span>
                  </label>
                </span>
              </p>
              <br />
            </div>
          </fieldset>
          <div
            class="field"
          >
            <label
              class="label "
              for="Interviewer ID"
            >
              Interviewer ID
            </label>
            <input
              class="input input--text input-type__input "
              id="Interviewer ID"
              name="Interviewer ID"
              type="text"
              value=""
            />
          </div>
          <div
            class="field"
          >
            <label
              class="label "
              for="Start date"
            >
              Start date
            </label>
            <input
              class="input input--text input-type__input "
              id="Start date"
              initial_value="2021-01-01"
              name="Start date"
              type="date"
              value="2021-01-01"
            />
          </div>
          <div
            class="field"
          >
            <label
              class="label "
              for="End date"
            >
              End date
            </label>
            <input
              class="input input--text input-type__input "
              id="End date"
              initial_value="2021-01-01"
              name="End date"
              type="date"
              value="2021-01-01"
            />
          </div>
          <br />
          <button
            class="btn "
            data-testid="submit-button"
            type="submit"
          >
            <span
              class="btn__inner"
            >
              Run
            </span>
          </button>
        </form>
        <br />
        <a
          download="interviewer-call-history-.csv"
          hidden=""
          href="data:text/csv;charset=utf-8,ï»¿\\"Interviewer\\",\\"Questionnaire\\",\\"Serial Number\\",\\"Call Start Time\\",\\"Call Length (Seconds)\\",\\"Call Result\\""
          target="_self"
        >
          Export report as Comma-Separated Values (CSV) file
        </a>
        <div
          class="panel panel--info panel--no-title  u-mt-m"
          hidden=""
        >
          <div
            class="panel__body "
          />
        </div>
        <br />
      </main>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="Breadcrumb"
      class="breadcrumb"
    >
      <ol
        class="breadcrumb__items u-fs-s"
      >
        <li
          class="breadcrumb__item"
          id="breadcrumb-0"
        >
          <a
            class="breadcrumb__link"
            href="/"
          >
            Back
          </a>
          <svg
            class="svg-icon"
            focusable="false"
            viewBox="0 0 8 13"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
              transform="translate(-5.02 -1.59)"
            />
          </svg>
        </li>
      </ol>
    </nav>
    <main
      class="page__main u-mt-s"
      id="main-content"
    >
      <h1
        class="u-mb-m"
      >
        Run interviewer call history report
      </h1>
      <p
        aria-live="polite"
        class="u-fs-s u-mt-s"
      >
        Data in this report was last updated: 
        <b>
          <time
            datetime="2000-01-01T10:00:00.000Z"
            title="2000-01-01 10:00"
          >
            2 days ago
          </time>
           (01/01/2000 10:00:00)
        </b>
      </p>
      <p
        class="u-fs-s"
      >
        Data in this report only goes back to the last 12 months.
      </p>
      <form
        action="#"
      >
        <fieldset
          class="fieldset"
        >
          <legend
            class="fieldset__legend"
          >
            Select survey
          </legend>
          <div
            class="radios__items"
            id="Survey TLA"
          >
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  checked=""
                  class="radio__input js-radio"
                  id="all"
                  initial_value="undefined"
                  name="Survey TLA"
                  type="radio"
                  value="undefined"
                />
                <label
                  class="radio__label "
                  for="all"
                  id="all-label"
                >
                  Show all surveys
                </label>
              </span>
            </p>
            <br />
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  class="radio__input js-radio"
                  id="lms"
                  initial_value="undefined"
                  name="Survey TLA"
                  type="radio"
                  value="lms"
                />
                <label
                  class="radio__label label--with-description"
                  for="lms"
                  id="lms-label"
                >
                  LMS
                  <span
                    class="label__description radio__label--with-description"
                    id="white-label-description-hint"
                  >
                    Labour Market Survey
                  </span>
                </label>
              </span>
            </p>
            <br />
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  class="radio__input js-radio"
                  id="opn"
                  initial_value="undefined"
                  name="Survey TLA"
                  type="radio"
                  value="opn"
                />
                <label
                  class="radio__label label--with-description"
                  for="opn"
                  id="opn-label"
                >
                  OPN
                  <span
                    class="label__description radio__label--with-description"
                    id="white-label-description-hint"
                  >
                    Opinions and Lifestyle Survey
                  </span>
                </label>
              </span>
            </p>
            <br />
          </div>
        </fieldset>
        <div
          class="field"
        >
          <label
            class="label "
            for="Interviewer ID"
          >
            Interviewer ID
          </label>
          <input
            class="input input--text input-type__input "
            id="Interviewer ID"
            name="Interviewer ID"
            type="text"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <label
            class="label "
            for="Start date"
          >
            Start date
          </label>
          <input
            class="input input--text input-type__input "
            id="Start date"
            initial_value="2021-01-01"
            name="Start date"
            type="date"
            value="2021-01-01"
          />
        </div>
        <div
          class="field"
        >
          <label
            class="label "
            for="End date"
          >
            End date
          </label>
          <input
            class="input input--text input-type__input "
            id="End date"
            initial_value="2021-01-01"
            name="End date"
            type="date"
            value="2021-01-01"
          />
        </div>
        <br />
        <button
          class="btn "
          data-testid="submit-button"
          type="submit"
        >
          <span
            class="btn__inner"
          >
            Run
          </span>
        </button>
      </form>
      <br />
      <a
        download="interviewer-call-history-.csv"
        hidden=""
        href="data:text/csv;charset=utf-8,ï»¿\\"Interviewer\\",\\"Questionnaire\\",\\"Serial Number\\",\\"Call Start Time\\",\\"Call Length (Seconds)\\",\\"Call Result\\""
        target="_self"
      >
        Export report as Comma-Separated Values (CSV) file
      </a>
      <div
        class="panel panel--info panel--no-title  u-mt-m"
        hidden=""
      >
        <div
          class="panel__body "
        />
      </div>
      <br />
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`interviewer call history report without data matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
      >
        <ol
          class="breadcrumb__items u-fs-s"
        >
          <li
            class="breadcrumb__item"
            id="breadcrumb-0"
          >
            <a
              class="breadcrumb__link"
              href="/"
            >
              Back
            </a>
            <svg
              class="svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
        </ol>
      </nav>
      <main
        class="page__main u-mt-s"
        id="main-content"
      >
        <h1
          class="u-mb-m"
        >
          Run interviewer call history report
        </h1>
        <p
          aria-live="polite"
          class="u-fs-s u-mt-s"
        >
          Data in this report was last updated: 
          <b>
            Invalid Date
          </b>
        </p>
        <p
          class="u-fs-s"
        >
          Data in this report only goes back to the last 12 months.
        </p>
        <form
          action="#"
        >
          <fieldset
            class="fieldset"
          >
            <legend
              class="fieldset__legend"
            >
              Select survey
            </legend>
            <div
              class="radios__items"
              id="Survey TLA"
            >
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    checked=""
                    class="radio__input js-radio"
                    id="all"
                    initial_value="undefined"
                    name="Survey TLA"
                    type="radio"
                    value="undefined"
                  />
                  <label
                    class="radio__label "
                    for="all"
                    id="all-label"
                  >
                    Show all surveys
                  </label>
                </span>
              </p>
              <br />
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    class="radio__input js-radio"
                    id="lms"
                    initial_value="undefined"
                    name="Survey TLA"
                    type="radio"
                    value="lms"
                  />
                  <label
                    class="radio__label label--with-description"
                    for="lms"
                    id="lms-label"
                  >
                    LMS
                    <span
                      class="label__description radio__label--with-description"
                      id="white-label-description-hint"
                    >
                      Labour Market Survey
                    </span>
                  </label>
                </span>
              </p>
              <br />
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    class="radio__input js-radio"
                    id="opn"
                    initial_value="undefined"
                    name="Survey TLA"
                    type="radio"
                    value="opn"
                  />
                  <label
                    class="radio__label label--with-description"
                    for="opn"
                    id="opn-label"
                  >
                    OPN
                    <span
                      class="label__description radio__label--with-description"
                      id="white-label-description-hint"
                    >
                      Opinions and Lifestyle Survey
                    </span>
                  </label>
                </span>
              </p>
              <br />
            </div>
          </fieldset>
          <div
            class="field"
          >
            <label
              class="label "
              for="Interviewer ID"
            >
              Interviewer ID
            </label>
            <input
              class="input input--text input-type__input "
              id="Interviewer ID"
              name="Interviewer ID"
              type="text"
              value=""
            />
          </div>
          <div
            class="field"
          >
            <label
              class="label "
              for="Start date"
            >
              Start date
            </label>
            <input
              class="input input--text input-type__input "
              id="Start date"
              initial_value="2000-01-03"
              name="Start date"
              type="date"
              value="2000-01-03"
            />
          </div>
          <div
            class="field"
          >
            <label
              class="label "
              for="End date"
            >
              End date
            </label>
            <input
              class="input input--text input-type__input "
              id="End date"
              initial_value="2000-01-03"
              name="End date"
              type="date"
              value="2000-01-03"
            />
          </div>
          <br />
          <button
            class="btn "
            data-testid="submit-button"
            type="submit"
          >
            <span
              class="btn__inner"
            >
              Run
            </span>
          </button>
        </form>
        <br />
        <a
          download="interviewer-call-history-.csv"
          hidden=""
          href="data:text/csv;charset=utf-8,ï»¿\\"Interviewer\\",\\"Questionnaire\\",\\"Serial Number\\",\\"Call Start Time\\",\\"Call Length (Seconds)\\",\\"Call Result\\""
          target="_self"
        >
          Export report as Comma-Separated Values (CSV) file
        </a>
        <div
          class="panel panel--info panel--no-title  u-mt-m"
          hidden=""
        >
          <div
            class="panel__body "
          />
        </div>
        <br />
      </main>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="Breadcrumb"
      class="breadcrumb"
    >
      <ol
        class="breadcrumb__items u-fs-s"
      >
        <li
          class="breadcrumb__item"
          id="breadcrumb-0"
        >
          <a
            class="breadcrumb__link"
            href="/"
          >
            Back
          </a>
          <svg
            class="svg-icon"
            focusable="false"
            viewBox="0 0 8 13"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
              transform="translate(-5.02 -1.59)"
            />
          </svg>
        </li>
      </ol>
    </nav>
    <main
      class="page__main u-mt-s"
      id="main-content"
    >
      <h1
        class="u-mb-m"
      >
        Run interviewer call history report
      </h1>
      <p
        aria-live="polite"
        class="u-fs-s u-mt-s"
      >
        Data in this report was last updated: 
        <b>
          Invalid Date
        </b>
      </p>
      <p
        class="u-fs-s"
      >
        Data in this report only goes back to the last 12 months.
      </p>
      <form
        action="#"
      >
        <fieldset
          class="fieldset"
        >
          <legend
            class="fieldset__legend"
          >
            Select survey
          </legend>
          <div
            class="radios__items"
            id="Survey TLA"
          >
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  checked=""
                  class="radio__input js-radio"
                  id="all"
                  initial_value="undefined"
                  name="Survey TLA"
                  type="radio"
                  value="undefined"
                />
                <label
                  class="radio__label "
                  for="all"
                  id="all-label"
                >
                  Show all surveys
                </label>
              </span>
            </p>
            <br />
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  class="radio__input js-radio"
                  id="lms"
                  initial_value="undefined"
                  name="Survey TLA"
                  type="radio"
                  value="lms"
                />
                <label
                  class="radio__label label--with-description"
                  for="lms"
                  id="lms-label"
                >
                  LMS
                  <span
                    class="label__description radio__label--with-description"
                    id="white-label-description-hint"
                  >
                    Labour Market Survey
                  </span>
                </label>
              </span>
            </p>
            <br />
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  class="radio__input js-radio"
                  id="opn"
                  initial_value="undefined"
                  name="Survey TLA"
                  type="radio"
                  value="opn"
                />
                <label
                  class="radio__label label--with-description"
                  for="opn"
                  id="opn-label"
                >
                  OPN
                  <span
                    class="label__description radio__label--with-description"
                    id="white-label-description-hint"
                  >
                    Opinions and Lifestyle Survey
                  </span>
                </label>
              </span>
            </p>
            <br />
          </div>
        </fieldset>
        <div
          class="field"
        >
          <label
            class="label "
            for="Interviewer ID"
          >
            Interviewer ID
          </label>
          <input
            class="input input--text input-type__input "
            id="Interviewer ID"
            name="Interviewer ID"
            type="text"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <label
            class="label "
            for="Start date"
          >
            Start date
          </label>
          <input
            class="input input--text input-type__input "
            id="Start date"
            initial_value="2000-01-03"
            name="Start date"
            type="date"
            value="2000-01-03"
          />
        </div>
        <div
          class="field"
        >
          <label
            class="label "
            for="End date"
          >
            End date
          </label>
          <input
            class="input input--text input-type__input "
            id="End date"
            initial_value="2000-01-03"
            name="End date"
            type="date"
            value="2000-01-03"
          />
        </div>
        <br />
        <button
          class="btn "
          data-testid="submit-button"
          type="submit"
        >
          <span
            class="btn__inner"
          >
            Run
          </span>
        </button>
      </form>
      <br />
      <a
        download="interviewer-call-history-.csv"
        hidden=""
        href="data:text/csv;charset=utf-8,ï»¿\\"Interviewer\\",\\"Questionnaire\\",\\"Serial Number\\",\\"Call Start Time\\",\\"Call Length (Seconds)\\",\\"Call Result\\""
        target="_self"
      >
        Export report as Comma-Separated Values (CSV) file
      </a>
      <div
        class="panel panel--info panel--no-title  u-mt-m"
        hidden=""
      >
        <div
          class="panel__body "
        />
      </div>
      <br />
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
